app/cdap/components/WrangleHome/Components/OngoingDataExplorations/index.tsx

```diff
@@ -14,113 +14,173 @@
  * the License.
  */
 
-import React, { useEffect, useState } from 'react';
 import { Box } from '@material-ui/core/';
-import { generateDataForExplorationCard } from './utils';
 import MyDataPrepApi from 'api/dataprep';
+import { getCategorizedConnections } from 'components/Connections/Browser/SidePanel/apiHelpers';
+import {
+  IConnectionWithConnectorType,
+  IExistingExplorationCardsData,
+  IConnectionsList,
+  IMassagedObject,
+  IValues,
+} from 'components/WrangleHome/Components/OngoingDataExplorations/types';
+import { getUpdatedExplorationCards } from 'components/WrangleHome/Components/OngoingDataExplorations/utils';
+import OngoingDataExplorationsCard from 'components/WrangleHome/Components/OngoingDataExplorationsCard';
+import T from 'i18n-react';
+import { orderBy } from 'lodash';
+import React, { useCallback, useEffect, useState } from 'react';
 import { Link } from 'react-router-dom';
-import { getCurrentNamespace } from 'services/NamespaceStore';
-import OngoingDataExplorationsCard from '../OngoingDataExplorationsCard';
-import { switchMap } from 'rxjs/operators';
 import { forkJoin } from 'rxjs/observable/forkJoin';
-import { IResponseData } from './types';
-import T from 'i18n-react';
+import { of } from 'rxjs/observable/of';
+import { defaultIfEmpty, switchMap } from 'rxjs/operators';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+
+export default function() {
+  const [onGoingExplorationsData, setOnGoingExplorationsData] = useState<IMassagedObject[]>([]);
 
-export default function OngoingDataExplorations() {
-  const [ongoingExpDatas, setOngoingExpDatas] = useState([]);
-  const [finalArray, setFinalArray] = useState([]);
+  const getOngoingData = useCallback(async () => {
+    const connectionsWithConnectorTypeData: Map<
+      string,
+      IConnectionsList[]
+    > = await getCategorizedConnections();
+    const connectionsWithConnectorTypeDataObject: IConnectionWithConnectorType[] = [];
+    for (const connectorName of connectionsWithConnectorTypeData.keys()) {
+      const values = connectionsWithConnectorTypeData.get(connectorName);
+      const connections = values.map((eachValue) => {
+        return {
```

Comment:
why do we need return statement, just do the following

    const connections = values.map((eachValue) => ({
          name: eachValue.name,
          connectorType: eachValue.connectionType,
})